src/lib/core/fiber.c

```diff
@@ -1498,6 +1516,35 @@ cord_create(struct cord *cord, const char *name)
 #endif
 }
 
+/**
+ * Delete the latest garbage fiber which couldn't be deleted somewhy before. Can
+ * safely rely on the fiber being not the current one. Because if it was added
+ * here before, it means some previous fiber put itself here, then died
+ * immediately afterwards for good, and gave control to another fiber. It
+ * couldn't be scheduled again.
+ */
+void
+cord_collect_garbage(struct cord *cord)
+{
+	struct fiber *garbage = cord->garbage;
+	if (garbage == NULL)
+		return;
+	assert(cord->fiber != garbage);
```

Comment:
Ok. I deleted it, moved `fiber_count_total()` into tests, and also I dropped one duplicated comment - it was stated both in .h and .c files.
<details><summary>diff</summary>

```Diff
diff --git a/src/lib/core/fiber.c b/src/lib/core/fiber.c
index c9f5d0efc..08087cfe4 100644
--- a/src/lib/core/fiber.c
+++ b/src/lib/core/fiber.c
@@ -1512,20 +1512,12 @@ cord_create(struct cord *cord, const char *name)
 #endif
 }
 
-/**
- * Delete the latest garbage fiber which couldn't be deleted somewhy before. Can
- * safely rely on the fiber being not the current one. Because if it was added
- * here before, it means some previous fiber put itself here, then died
- * immediately afterwards for good, and gave control to another fiber. It
- * couldn't be scheduled again.
- */
 void
 cord_collect_garbage(struct cord *cord)
 {
 	struct fiber *garbage = cord->garbage;
 	if (garbage == NULL)
 		return;
-	assert(cord->fiber != garbage);
 	fiber_delete(cord, garbage);
 	cord->garbage = NULL;
 }
@@ -1874,11 +1866,3 @@ int fiber_stat(fiber_stat_cb cb, void *cb_ctx)
 	}
 	return 0;
 }
-
-int
-fiber_count_total(void)
-{
-	size_t res = mempool_count(&cord()->fiber_mempool);
-	assert(res <= INT_MAX);
-	return (int)res;
-}
diff --git a/src/lib/core/fiber.h b/src/lib/core/fiber.h
index 4cb02248e..305255c9a 100644
--- a/src/lib/core/fiber.h
+++ b/src/lib/core/fiber.h
@@ -887,10 +887,6 @@ typedef int (*fiber_stat_cb)(struct fiber *f, void *ctx);
 int
 fiber_stat(fiber_stat_cb cb, void *cb_ctx);
 
-/** Total count of allocated fibers in the cord. Including dead ones. */
-int
-fiber_count_total(void);
-
 #if ENABLE_FIBER_TOP
 bool
 fiber_top_is_enabled(void);
diff --git a/test/unit/fiber.cc b/test/unit/fiber.cc
index 8290cd34d..c0a46c612 100644
--- a/test/unit/fiber.cc
+++ b/test/unit/fiber.cc
@@ -11,6 +11,15 @@ static struct fiber_attr default_attr;
 static unsigned long page_size;
 #define PAGE_4K 4096
 
+/** Total count of allocated fibers in the cord. Including dead ones. */
+static int
+fiber_count_total(void)
+{
+	size_t res = mempool_count(&cord()->fiber_mempool);
+	assert(res <= INT_MAX);
+	return (int)res;
+}
+
 static int
 noop_f(va_list ap)
 {
diff --git a/test/unit/fiber_stack.c b/test/unit/fiber_stack.c
index 72c3beaef..5ab1f3358 100644
--- a/test/unit/fiber_stack.c
+++ b/test/unit/fiber_stack.c
@@ -6,6 +6,15 @@
 
 static struct fiber_attr default_attr;
 
+/** Total count of allocated fibers in the cord. Including dead ones. */
+static int
+fiber_count_total(void)
+{
+	size_t res = mempool_count(&cord()->fiber_mempool);
+	assert(res <= INT_MAX);
+	return (int)res;
+}
+
 static int
 noop_f(va_list ap)
 {
```
</details>

Reply:
thanks!